app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/services/notifications/FirebaseNotificationService.java

```diff
@@ -29,19 +43,46 @@ public Task<Void> subscribeToUserMealPosts(String uid) {
     }
     @Override
     public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
+        createNotificationChannel();
+
         String mealId = remoteMessage.getData().get("mealId");
-        System.out.println(mealId);
         String vendorID = remoteMessage.getData().get("vendorID");
-        System.out.println(vendorID);
-        new Handler(Looper.getMainLooper()).post(() -> {
-            Toast toast = Toast.makeText(getApplicationContext(),
-                    "New meal posted by "+vendorID+": "+mealId,
-                    Toast.LENGTH_LONG);
+        AppContainer appContainer = ((NeighborFoodApplication) this.getApplication()).getAppContainer();
+        /*
+        I couldn't fetch the data from the database that's why I had to remove the following two lines
+
+        User vendor = appContainer.getUserRepo().getUserById(vendorID).getResult();
```

Comment:
getUserById is an asynchronous function, so getting the result will generate an illegal State Exception when you try to get the result that is not finished. We can chain tasks to get the user and meal from the repository, then add a callback to launch the notification when the tasks complete